source/fab/tools/categories.py

```diff
@@ -0,0 +1,37 @@
+##############################################################################
+# (c) Crown copyright Met Office. All rights reserved.
+# For further details please refer to the file COPYRIGHT
+# which you should have received as part of this distribution
+##############################################################################
+
+'''This simple module defines an Enum for all allowed categories.
+'''
+
+from enum import auto, Enum
+
+
+class Categories(Enum):
+    '''This class defines the allowed tool categories.'''
+
+    C_COMPILER = auto()
+    C_PREPROCESSOR = auto()
+    FORTRAN_COMPILER = auto()
+    FORTRAN_PREPROCESSOR = auto()
+    LINKER = auto()
+    PSYCLONE = auto()
+    FCM = auto()
+    GIT = auto()
+    SUBVERSION = auto()
+    AR = auto()
+    RSYNC = auto()
+    MISC = auto()
+
+    def __str__(self):
+        '''Simplify the str output by using only the name (e.g. `C_COMPILER`
+        instead of `Categories.C_COMPILER)`.'''
+        return str(self.name)
+
+    @property
+    def is_compiler(self):
```

Comment:
Because the ToolRepository adds automatically each compiler as linker. While this could just check the Tool's Categories of course, it would be limited to compilers that are specified in the Categories, while this way a user *could* add a new compiler as MISC tool, and would still get a linker for that new language.

But tbh, I don't see this as a real use case ;)

I might change this as part of #312. One option could also be to rename this function to `can_be_used_as_linker`?